API/API.mli

```diff
@@ -5666,7 +5666,7 @@ module Vernacinterp :
 sig
   type deprecation = bool
 
-  type vernac_command = Genarg.raw_generic_argument list -> unit -> unit
+  type vernac_command = Genarg.raw_generic_argument list -> Loc.t option -> unit
```

Comment:
@herbelin IMHO this is orthogonal, the thunk could be kept in both approaches. @matejkosik the real question here to know what we should use is "is the first argument a located object"? If so, IMHO, we should then use the existing abstraction.

Note of course that not all objects involving locations do indeed meet this criteria.